aiida/backends/sqlalchemy/models/node.py

```diff
@@ -43,18 +44,16 @@ class DbNode(Base):
     __tablename__ = 'db_dbnode'
 
     id = Column(Integer, primary_key=True)  # pylint: disable=invalid-name
-    uuid = Column(UUID(as_uuid=True), default=get_new_uuid, unique=True)
-    node_type = Column(String(255), index=True)
-    process_type = Column(String(255), index=True)
-    label = Column(
-        String(255), index=True, nullable=True, default=''
-    )  # Does it make sense to be nullable and have a default?
-    description = Column(Text(), nullable=True, default='')
-    ctime = Column(DateTime(timezone=True), default=timezone.now)
-    mtime = Column(DateTime(timezone=True), default=timezone.now, onupdate=timezone.now)
+    uuid = Column(UUID(as_uuid=True), default=get_new_uuid, nullable=False)
+    node_type = Column(String(255), default='', nullable=False)
+    process_type = Column(String(255))
+    label = Column(String(255), nullable=False, default='')
+    description = Column(Text(), nullable=False, default='')
+    ctime = Column(DateTime(timezone=True), default=timezone.now, nullable=False)
+    mtime = Column(DateTime(timezone=True), default=timezone.now, onupdate=timezone.now, nullable=False)
     attributes = Column(JSONB)
     extras = Column(JSONB)
-    repository_metadata = Column(JSONB, nullable=False, default=dict, server_default='{}')
+    repository_metadata = Column(JSONB, nullable=True, default=dict)
```

Comment:
I remember adding this explicitly. The commit message (4a347b6001fc4f4dcc1c0710489a899b6589ee57) explains why. I think it makes sense to enforce that this column always has an empty dictionary. It prevented a bug. Why can we now revert this?

Reply:
Because there is not a matching server default in the django schema (it is not possible: https://code.djangoproject.com/ticket/470). The `server_default` did not fix the bug in 4a347b6, it was the addition of `default=dict` on the python side.

Reply:
Indeed though, ideally we would make more use of server-side defaults, which will be possible once we don't have to deal with the shortcomings of django 😉

Reply:
Ok not ideal that we have to adapt to Django if that is what we are going to drop, but if that's the way it is. Maybe we can optimize this once Django is dropped and add a migration for SqlAlchemy to do things as they should be ideally